compiler/prelude/primops.txt.pp

```diff
@@ -818,8 +818,13 @@
 
 primop  IndexArrayOp "indexArray#" GenPrimOp
    Array# a -> Int# -> (# a #)
-   {Read from specified index of immutable array. Result is packaged into
-    an unboxed singleton; the result itself is not yet evaluated.}
+   {Read from specified index of immutable array. Result is packaged
+    into an unboxed unary tuple; the result itself is not yet
+    evaluated. Pattern matching on the tuple forces the indexing of the
+    array to happen but does not evaluate the element itself. Evaluating
+    the thunk prevents additional thunks from building up on the
```

Comment:
But that's not right either. It's the `{\tt (# a #)}`. Would writing it like that show up correctly. Also, perhaps I should mention how an unboxed unary tuple is evaluated, since it's an uncommon construct. Something like:

> Evaluating the `(# a #)` (by casing on it or using a let binding to extract the inner value) prevents ...

Reply:
The word "extract" is awful though. Maybe something more like "bind an identifier to the inner value".